docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+2. You push the changes to your public repository and publish links to your changes in the Moodle Tracker.
+3. You request a peer review of your code from another developer.
+4. When the peer reviewer is happy, they submit the issue for integration.
+5. Moodle integrators pull the changes from your public repository, and if approved, they merge them into Moodle's integration repository.
+6. The integrated change are tested and finally pushed into Moodle's production repository.
+7. You syncronise your local repository with Moodle's production repository and the process repeats.
+
+This workflow runs in weekly cycles. The integration happens on Monday and Tuesday and the testing on Wednesday. On Thursday (or Friday if testing takes too long), the production repository (moodle.git) is usually updated with changes from the last development week.
+
+Most Moodle developers have their public repositories hosted at [Github](http://github.com/). In these examples, we'll assume you've set up your public repository there too.
+
+:::note
+When you first register on Moodle Tracker you can not assign issues to yourself or send them for peer review. You will be added to the developers group after your first bug fix is integrated. Before that, just comment on the issue with a link to your branch and a component lead, or another developer, will send the issue for peer review for you.
+:::
+
+## Installing Git on your computer
+
+### Installing Git
+
+Most Linux distributions have Git available as a package to install.
+
+#### Debian/Ubuntu
+
+```
+sudo apt-get install git
+```
+
+#### MacOS (Homebrew)
+
+```
+brew install git
+```
+
+#### Windows
+
+Download the latest Git installer at [Git for Windows](https://gitforwindows.org/).
+
+## Configuring Git
+
+After installation, set your name and email. Your name and email will become part of your commits and can't be changed later once your commits are accepted into Moodle code. Therefore we ask contributors to use their real names written in with proper capitalisation. E.g. "John Smith" and not "john smith", or even "john5677".
+
+```
+    git config --global user.name "Your Name"
+    git config --global user.email yourmail@domain.tld
+```
+
+Unless you are the repository maintainer, it is wise to set your Git to not push changes in file permissions:
+
+```
+    git config --global core.filemode false
+```
+
+It's recommended to verify that the your Git installation is not performing any transformation between LFs and CRLFs. Moodle use **only LFs** and you should configure your editor/IDE the same way. Note that having any "magic" enabled is known to cause [problems with unit tests](https://docs.moodle.org/dev/Common_unit_test_problems#The_test_file_.22evolution.test.22_should_not_contain_section_named_.22.5Blots_of_content.5D.22) execution. We recommend you to set:
+
+```
+    git config --global core.autocrlf false
+```
+
+## Setting-up your public repository
+
```

Comment:
Maybe worth adding a note here that you can use any Git repository host, including self-hosted. However, most people use GitHub because it is free, and that Moodle incorporates a series of GitHub Actions to perform common checks with your patch. yaddah yaddah.

Reply:
Good suggestion. I will add that in.